server/src/test/java/org/opensearch/gateway/ReplicaShardBatchAllocatorTests.java

```diff
@@ -634,6 +643,50 @@ public void testDoNotCancelForBrokenNode() {
         assertThat(allocation.routingNodes().shardsWithState(ShardRoutingState.UNASSIGNED), empty());
     }
 
+    public void testAllocateUnassignedBatchThrottlingAllocationDeciderIsHonoured() throws InterruptedException {
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        AllocationDeciders allocationDeciders = randomAllocationDeciders(
+            Settings.builder()
+                .put(ThrottlingAllocationDecider.CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_RECOVERIES_SETTING.getKey(), 1)
+                .put(ThrottlingAllocationDecider.CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_INCOMING_RECOVERIES_SETTING.getKey(), 1)
+                .build(),
+            clusterSettings,
+            random()
+        );
+        setUpShards(2);
+        final RoutingAllocation routingAllocation = twoAssignedPrimaryShardsAnd1UnassignedReplicaPerShard(allocationDeciders);
+        for (ShardId shardIdFromBatch : shardsInBatch) {
+            testBatchAllocator.addShardData(
+                node1,
+                shardIdFromBatch,
+                "MATCH",
+                null,
+                new StoreFileMetadata("file1", 10, "MATCH_CHECKSUM", MIN_SUPPORTED_LUCENE_VERSION)
+            )
+                .addShardData(
+                    node2,
+                    shardIdFromBatch,
+                    "NO_MATCH",
+                    null,
+                    new StoreFileMetadata("file1", 10, "MATCH_CHECKSUM", MIN_SUPPORTED_LUCENE_VERSION)
+                )
+                .addShardData(
+                    node3,
+                    shardIdFromBatch,
+                    "MATCH",
+                    null,
+                    new StoreFileMetadata("file1", 10, "MATCH_CHECKSUM", MIN_SUPPORTED_LUCENE_VERSION)
+                );
+        }
+        allocateAllUnassignedBatch(routingAllocation);
+        // Verify the throttling decider was throttled, incoming recoveries on a node should be
+        // lesser than or equal to allowed concurrent recoveries
```

Comment:
Can we verify the Unassigned Reason of the shard for the throttling ?

Reply:
The Unassigned reason for the shard is not updated for replicas shards that are ignored. 
For our test case unassigned shard was created with NO_ATTEMPT status, the status doesn't change but its marked is as ignored after the delayed allocation run.